### PR TITLE
Add in VUE date picker component

### PR DESCRIPTION
**Pull request by drummond-dds**
This component adds a VUE.js date picker that follows the [U.S. Web Design System style guide](https://designsystem.digital.gov/components/form-controls/#date-input). Takes in optional `mindate` and `maxdate` to throw error messages when validation fails.

* [Pivotal Tracker](https://www.pivotaltracker.com/story/show/162771867)

---

jan-25-2019 14-15-46 gif sb-d95c8a9f-z5r6pr

### CODE REVIEW

File: templates/components/date_picker.html [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  initialday="{{ field.data.day }}"
+  initialyear="{{ field.data.year }}"
+  inline-template>
+
+    <div class="usa-input date-picker" v-bind:class="{ 'usa-input--success': isDateValid }">
+      <p v-if="!isWithinDateRange" class="usa-input-error-message">
+        {% if maxdate and mindate %}Date must be between {{maxdate.strftime("%Y-%m-%d")}} and {{mindate.strftime("%Y-%m-%d")}}{% endif %}
+        {% if maxdate and not mindate %}Date must be before or on {{maxdate.strftime("%Y-%m-%d")}}{% endif %}
+        {% if mindate and not maxdate %}Date must be after or on {{mindate.strftime("%Y-%m-%d")}}{% endif %}
+      </p>
```
Comment (dandds) [inline templates/components/date_picker.html]: I'm not sure about the placement of this error message; it always shifts the input down when there's an error. Do we have any examples from the wireframes?

File: templates/components/date_picker.html [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            {% if maxdate %}max="{{ maxdate.year }}"{% endif %}
+            {% if mindate %}min="{{ mindate.year }}"{% endif %}
+          />
+        </div>
+
+        <div class="usa-form-group-date-ok" v-if="isDateValid">
+          {{ Icon("ok", classes="icon--green") }}
+        </div>
+
+        <input name="{{ field.name }}" v-model="formattedDate" type="hidden" />
```
Comment (dandds) [inline templates/components/date_picker.html]: I might be missing something, but why are there two hidden inputs (top and bottom of the div)?

File: styles/components/_forms.scss [hunk 3]
```diff
@@ -116,3 +116,25 @@
...
+  display: inline-block;
+  margin-top: 10px;
+  width: 100%;
+
+  label {
+    font-weight: 400;
+  }
+
+  input[type="number"]::-webkit-inner-spin-button {
+    appearance: none;
```
Comment (dandds) [inline styles/components/_forms.scss]: This doesn't work for me in FF 64.

Comment (dandds) [reply styles/components/_forms.scss]: screen shot 2019-01-28 at 9 31 59 am

File: templates/components/date_picker.html [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+            v-bind:max="daysMaxCalculation"
+            v-model="day"
+          />
+        </div>
+
+        <div class="usa-form-group usa-form-group-year">
+          <label>Year</label>
+          <input
+            id="date-year"
+            maxlength="2"
```
Comment (dandds) [inline templates/components/date_picker.html]: This should be 4, right? Interestingly, FF ignored this but IE10 was a stickler about it.

File: js/components/date_selector.js [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+        this.month &&
+        this.year &&
+        this.isDayValid &&
+        this.isMonthValid &&
+        this.isYearValid &&
+        this.isWithinDateRange
+      )
+    },
+
+    daysMaxCalculation: function() {
```
[the remark this answers is not in the document]
Comment (drummond-dds) [reply js/components/date_selector.js]: @patricksmithdds I added in logic for handling leap years

File: templates/components/date_picker.html (hunk 2 above)
Comment (drummond-dds) [reply templates/components/date_picker.html]: Whoops - a typo. Fixed in 1143a34b

File: templates/components/date_picker.html (hunk 1 above)
Comment (drummond-dds) [reply templates/components/date_picker.html]: I can't find anything in the wireframes but I just pushed up c8ffb4e15060297a57e862742bd55e7c723521f4 which moves it to below the input (which is how we currently do it).

screen shot 2019-01-28 at 10 48 56

**Pull request merged**